md-dao/src/main/java/com/wipro/ats/bdre/md/dao/InstanceExecDAO.java

```diff
@@ -160,4 +164,56 @@ public void delete(Long id) {
             session.close();
         }
     }
+
+
+   public List<SLAMonitoringBean> slaMonitoringData(List<Process> subProcessList)
+    {
+        Session session = sessionFactory.openSession();
+        List<SLAMonitoringBean> slaMonitoringBeanList=new ArrayList<>();
+        try {
+            session.beginTransaction();
+            for(Process process:subProcessList)
+            {
+               Criteria instanceExecListCriteria= session.createCriteria(InstanceExec.class).add(Restrictions.eq("process",process)).addOrder(Order.asc("startTs")).setMaxResults(25);
+                List<InstanceExec> instanceExecList=instanceExecListCriteria.list();
+                long sumTime=0;
+                long currentTime=0;
+                int total=instanceExecList.size();
+                for (InstanceExec instanceExec:instanceExecList)
+                {
+                    if(instanceExec.getEndTs()!=null){
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)